### PR TITLE
Fixes #9564 - Document module version format

### LINKED ISSUE #9564: [PowershellGet] Add the requirements on the Prerelease string
### Prerequisites

- [X] **Existing Issue:** Search the existing issues for this repository. If there is an issue that fits your needs do not file a new one. Subscribe, react, or comment on that issue instead.
- [X] **Descriptive Title:** Write the title for this issue as a short synopsis. If possible, provide context. For example, "Document new `Get-Foo` cmdlet" instead of "New cmdlet."

### PowerShell Version

5.1, 7.0, 7.2, 7.3

### Summary

The information given in this [link](https://learn.microsoft.com/en-us/powershell/scripting/gallery/concepts/module-prerelease-support?view=powershell-5.1) should be added to the documentation of cmdlets using a version number ( Save -Module , etc).

### Details

The type of parameters referencing a version number is 'String' but it is unclear which syntax is allowed.

### Proposed Content Type

Cmdlet Reference, About Topic, Concept

### Proposed Title

_No response_

### Related Articles

https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/5.1/PowershellGet/Save-Module.md
etc.

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#53168](https://dev.azure.com/msft-skilling/cebd7ef5-4282-448b-9701-88c8637581b7/_workitems/edit/53168) - Fixes #9564 - Document module version format

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: reference/5.1/PowershellGet/Find-Module.md [hunk 1]
```diff
@@ -49,6 +48,12 @@ If the **RequiredVersion** parameter is specified, `Find-Module` only returns th
 that is an exact match to the specified version. `Find-Module` searches through all available
 modules, because name conflicts between sources can occur.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Find-Module.md:51]:
```suggestion
The parameters that take module version numbers take strings formatted as version numbers.
```

Comment (michaeltlombardi) [reply reference/5.1/PowershellGet/Find-Module.md:51]: Nit: Not sure about "the parameters that take... take..."

File: reference/5.1/PowershellGet/Find-Module.md [hunk 2]
```diff
@@ -49,6 +48,12 @@ If the **RequiredVersion** parameter is specified, `Find-Module` only returns th
...
+
+- Standard version numbers have a format of `x.y.z` where x, y, and z are numbers
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Find-Module.md:53]: Does it not support the four-digit version schemes, like `1.2.3.4`? Or is that what's meant by "standard" here? Same comment for two-digit schemes.

File: reference/5.1/PowershellGet/Install-Module.md [hunk 3]
```diff
@@ -44,6 +44,12 @@ module, warning messages are displayed. After you confirm that you want to insta
 override the warnings, use the `-Force` and `-AllowClobber` parameters. Dependent upon your
 repository settings, you might need to answer a prompt for the module installation to continue.
 
+The parameters that take module version number take strings formatted as version numbers.
+
+- Standard version numbers have a format of `x.y.z` where x, y, and z are numbers
+- Prerelease versions have a format of `x.y.z-<prerelease_label>` where the `<prerelease_label>` is
+  arbitrary string assigned to that release.
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Install-Module.md:51]: As above.

File: reference/5.1/PowershellGet/Install-Module.md [hunk 4]
```diff
@@ -113,12 +119,52 @@ The `Install-Module` uses the **Name** parameter to specify the **PowerShellGet*
...
+Install-Module -Name Microsoft.PowerShell.Crescendo -AllowPrerelease
+```
+
+Using this method you get the latest version available. If the latest version isn't a prerelease,
+you get the latest stable version of the module.
+
+### Example 7: Install a specific prerelease version of a module
+
+This example shows how to install a specific prerelease version of a module. The `Find-Module`
+cmdlet can be used to find prerelease versions of modules in the PowerShell Gallery.
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Install-Module.md:137]: Nit: Suggest rewording to "You can use the `Find-Module`..."

File: reference/5.1/PowershellGet/Install-Module.md [hunk 5]
```diff
@@ -324,8 +370,8 @@ Accept wildcard characters: False
 
 ### -Repository
 
-Use the **Repository** parameter to specify which repository is used to download and install a
-module. Used when multiple repositories are registered. Specifies the name of a registered
+Use the **Repository** parameter to specify the name of repository from which to download and
+install a module. Used when multiple repositories are registered. Specifies the name of a registered
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Install-Module.md:374]: Not sure on the first sentence. Maybe:

```suggestion
Use the **Repository** parameter to specify the name of the repository to download and
install a module from. Used when multiple repositories are registered. Specifies the name of a registered
```

File: reference/5.1/PowershellGet/Install-Module.md [hunk 6]
```diff
@@ -365,19 +411,19 @@ Accept wildcard characters: False
...
-The **CurrentUser** installs modules in a location that is accessible only to the current user of
+The **CurrentUser** installs modules in a location that's accessible only to the current user of
 the computer. For example:
 
 `$HOME\Documents\WindowsPowerShell\Modules`
 
 When no **Scope** is defined, the default is set based on the PowerShellGet version.
 
-- In PowerShellGet versions 2.0.0 and above, the default is **CurrentUser**, which does not require
+- In PowerShellGet versions 2.0.0 and above, the default is **CurrentUser**, which doesn't require
```
Comment (michaeltlombardi) [inline reference/5.1/PowershellGet/Install-Module.md:426]:
```suggestion
- In PowerShellGet version 2.0 and above, the default is **CurrentUser**, which doesn't require
```

**Review by michaeltlombardi — CHANGES_REQUESTED** [review]
Minor typo, suggestion, and some nits.

File: reference/5.1/PowershellGet/Find-Module.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/PowershellGet/Find-Module.md:53]: Don't know but not going to get into it here.

File: reference/5.1/PowershellGet/Install-Module.md [hunk 7]
```diff
@@ -44,6 +44,12 @@ module, warning messages are displayed. After you confirm that you want to insta
 override the warnings, use the `-Force` and `-AllowClobber` parameters. Dependent upon your
 repository settings, you might need to answer a prompt for the module installation to continue.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Install-Module.md:47]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/5.1/PowershellGet/Install-Module.md (hunk 4 above)
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Install-Module.md:137]:
```suggestion
This example shows how to install a specific prerelease version of a module. You can use the `Find-Module`
cmdlet to find prerelease versions of modules in the PowerShell Gallery.
```

File: reference/7.2/PowerShellGet/Save-Module.md [hunk 8]
```diff
@@ -60,6 +60,12 @@ inspection by an administrator. The saved module can then be copied into the app
 `Get-PSRepository` displays the local computer's registered repositories. You can use the
 `Find-Module` cmdlet to search registered repositories.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.2/PowerShellGet/Save-Module.md:63]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.2/PowerShellGet/Uninstall-Module.md [hunk 9]
```diff
@@ -38,6 +38,12 @@ uninstall a module if other modules depend on it or the module wasn't installed
 You can manually delete module files, but doing so may break any modules that depend on the deleted
 module.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.2/PowerShellGet/Uninstall-Module.md:41]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.2/PowerShellGet/Update-Module.md [hunk 10]
```diff
@@ -32,9 +32,15 @@ were installed on the local computer with `Install-Module`. `Update-Module` sear
 `$env:PSModulePath` for installed modules.
 
 `Update-Module` with no parameters specified updates all installed modules. To specify a module to
-update, use the **Name** parameter. You can update to a module's specific version by using the
+update, use the **Name** parameter. You can update to a module's specific version using the
 **RequiredVersion** parameter.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.2/PowerShellGet/Update-Module.md:38]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.3/PowerShellGet/Find-Module.md [hunk 11]
```diff
@@ -48,6 +48,12 @@ If the **RequiredVersion** parameter is specified, `Find-Module` only returns th
 that is an exact match to the specified version. `Find-Module` searches through all available
 modules, because name conflicts between sources can occur.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.3/PowerShellGet/Find-Module.md:51]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.3/PowerShellGet/Publish-Module.md [hunk 12]
```diff
@@ -51,6 +51,12 @@ from the module manifest, some metadata must be specified in `Publish-Module` pa
 **Tag**, **ReleaseNote**, **IconUri**, **ProjectUri**, and **LicenseUri**, because these parameters
 match fields in a NuGet-based gallery.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.3/PowerShellGet/Publish-Module.md:54]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.3/PowerShellGet/Save-Module.md [hunk 13]
```diff
@@ -60,6 +60,12 @@ inspection by an administrator. The saved module can then be copied into the app
 `Get-PSRepository` displays the local computer's registered repositories. You can use the
 `Find-Module` cmdlet to search registered repositories.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.3/PowerShellGet/Save-Module.md:63]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.3/PowerShellGet/Uninstall-Module.md [hunk 14]
```diff
@@ -38,6 +38,12 @@ uninstall a module if other modules depend on it or the module wasn't installed
 You can manually delete module files, but doing so may break any modules that depend on the deleted
 module.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.3/PowerShellGet/Uninstall-Module.md:41]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.3/PowerShellGet/Update-Module.md [hunk 15]
```diff
@@ -32,9 +32,15 @@ were installed on the local computer with `Install-Module`. `Update-Module` sear
 `$env:PSModulePath` for installed modules.
 
 `Update-Module` with no parameters specified updates all installed modules. To specify a module to
-update, use the **Name** parameter. You can update to a module's specific version by using the
+update, use the **Name** parameter. You can update to a module's specific version using the
 **RequiredVersion** parameter.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.3/PowerShellGet/Update-Module.md:38]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/5.1/PowershellGet/Publish-Module.md [hunk 16]
```diff
@@ -51,6 +51,12 @@ from the module manifest, some metadata must be specified in `Publish-Module` pa
 **Tag**, **ReleaseNote**, **IconUri**, **ProjectUri**, and **LicenseUri**, because these parameters
 match fields in a NuGet-based gallery.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Publish-Module.md:54]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/5.1/PowershellGet/Save-Module.md [hunk 17]
```diff
@@ -60,6 +60,12 @@ inspection by an administrator. The saved module can then be copied into the app
 `Get-PSRepository` displays the local computer's registered repositories. You can use the
 `Find-Module` cmdlet to search registered repositories.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Save-Module.md:63]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/5.1/PowershellGet/Uninstall-Module.md [hunk 18]
```diff
@@ -38,6 +38,12 @@ uninstall a module if other modules depend on it or the module wasn't installed
 You can manually delete module files, but doing so may break any modules that depend on the deleted
 module.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Uninstall-Module.md:41]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/5.1/PowershellGet/Update-Module.md [hunk 19]
```diff
@@ -31,9 +31,15 @@ were installed on the local computer with `Install-Module`. `Update-Module` sear
 `$env:PSModulePath` for installed modules.
 
 `Update-Module` with no parameters specified updates all installed modules. To specify a module to
-update, use the **Name** parameter. You can update to a module's specific version by using the
+update, use the **Name** parameter. You can update to a module's specific version using the
 **RequiredVersion** parameter.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/5.1/PowershellGet/Update-Module.md:37]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.2/PowerShellGet/Find-Module.md [hunk 20]
```diff
@@ -48,6 +48,12 @@ If the **RequiredVersion** parameter is specified, `Find-Module` only returns th
 that is an exact match to the specified version. `Find-Module` searches through all available
 modules, because name conflicts between sources can occur.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.2/PowerShellGet/Find-Module.md:51]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

File: reference/7.2/PowerShellGet/Install-Module.md [hunk 21]
```diff
@@ -44,6 +44,12 @@ module, warning messages are displayed. After you confirm that you want to insta
 override the warnings, use the `-Force` and `-AllowClobber` parameters. Dependent upon your
 repository settings, you might need to answer a prompt for the module installation to continue.
 
+The parameters that take module version number take strings formatted as version numbers.
```
Comment (sdwheeler) [inline reference/7.2/PowerShellGet/Install-Module.md:47]:
```suggestion
The parameters that take module version numbers expect strings formatted as version numbers.
```

**Review by michaeltlombardi — APPROVED** [review]
